test/e2e/invalid_meteringconfig_produces_error_tests.go

```diff
@@ -0,0 +1,23 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/operator-metering/test/reportingframework"
+)
+
+func testInvalidMeteringConfigMissingStorageSpec(t *testing.T, rf *reportingframework.ReportingFramework) {
+	require.NotNil(t, rf, "expected the reportingframework object would not be nil")
+	require.NotEmpty(t, rf.Namespace, "expected the reportingframework Namespace field would not be empty")
+	require.NotNil(t, rf.MeteringClient, "expected the reportingframework MeteringClient field would not be nil")
+
+	// we expect that the returned mc.Spec will only have nil fields (e.g. Spec:{DisableOCPFeatures:<nil> LogHelmTemplate:<nil>)
+	mc, err := rf.MeteringClient.MeteringConfigs(rf.Namespace).Get("operator-metering", meta.GetOptions{})
+	assert.Error(t, err, "expected there would be an error trying to query for the invalid meteringconfig")
```

Comment:
Maybe instead of assert.Error we verify the error is a kubernetes not found error. Since that's normally determined by a helper, we could do something like `assert.True(t, apierrors.IsNotFound(err), "expected MeteringConfig to not exist")`

Reply:
At one point that was in the previous implementation (in the consolidating PR before dropping those commits) but I must have applied the wrong stash - I agree that's a much better check that just asserting there was an error trying to query for the intended `MeteringConfig`.